### PR TITLE
Resolve azure provider deprecations in tests

### PR DESCRIPTION
**Pull request by dirrao**
Resolve azure provider deprecations in tests
related: https://github.com/apache/airflow/issues/38642

### CODE REVIEW

File: tests/providers/microsoft/azure/hooks/test_adx.py [hunk 1]
```diff
@@ -279,20 +279,28 @@ def test_run_query(self, mock_execute, mocked_connection):
...
         hook = AzureDataExplorerHook(azure_data_explorer_conn_id=mocked_connection.conn_id)
-        assert hook.connection._kcsb.data_source == "host"
+        if warning:
+            with pytest.warns(
+                AirflowProviderDeprecationWarning,
+                match="`extra__azure_data_explorer__.*` is deprecated in azure connection extra, please use `.*` instead",
+            ):
+                assert hook.connection._kcsb.data_source == "host"
+        else:
+            assert hook.connection._kcsb.data_source == "host"
```
Comment (eladkal) [inline tests/providers/microsoft/azure/hooks/test_adx.py]: I don't know if we need it?
https://github.com/apache/airflow/pull/27219/files suggets that that this was to preserve backward compatibility with min Airflow version 2.5.0 now that we min airflow version is 2.7.0 maybe we can safely remove the backported code?
cc @dstandish

Comment (dirrao) [reply tests/providers/microsoft/azure/hooks/test_adx.py]: Event I am not sure about it. @dstandish Can you confirm ^^?

Comment (dirrao) [reply tests/providers/microsoft/azure/hooks/test_adx.py]: @eladkal Shall I go ahead and make changes as per your suggestion or wait for @dstandish inputs?

Comment (eladkal) [reply tests/providers/microsoft/azure/hooks/test_adx.py]: If the code is for backport with airflow versions older than 2.7.0 then we can just remove it

Comment (dirrao) [reply tests/providers/microsoft/azure/hooks/test_adx.py]: I have updated the code.

**Comment by dirrao** [comment]
@eladkal 
I have updated the code. Can you re-review it once?

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Comment by dirrao** [comment]
@eladkal  / @potiuk 
Can you review and merge this MR?

**Pull request merged**